x/posts/internal/keeper/handler.go

```diff
@@ -35,6 +35,35 @@ func NewHandler(keeper Keeper) sdk.Handler {
 	}
 }
 
+// ValidatePost checks if the given post is valid according to the current posts' module params
+func ValidatePost(ctx sdk.Context, k Keeper, post types.Post) error {
+	params := k.GetParams(ctx)
+	maxMsgLen := params.MaxPostMessageLength.Int64()
+	maxOpFieldNum := params.MaxOptionalDataFieldsNumber.Int64()
+	maxOpFieldValLen := params.MaxOptionalDataFieldValueLength.Int64()
+
+	if len(post.Message) > int(maxMsgLen) {
```

Comment:
I think that for this line as well all the other lengths checks, it's better to convert the left side of the condition to `int64` instead of forcing the right one to `int`. This would avoid any exception if `maxMsgLen` is greater than the max `int` value allowed:


```golang
	if int64(len(post.Message)) > maxMsgLen {
```